tests/network/bindingplugin_passt.go

```diff
@@ -331,5 +336,86 @@ EOL`, inetSuffix, serverIP, serverPort)
 			By("Checking ping (IPv6) from VM to cluster nodes gateway")
 			Expect(libnet.PingFromVMConsole(vmi, ipv6Address)).To(Succeed())
 		})
+
+		DescribeTable("Connectivity should be preserved after migration", func(ipFamily k8sv1.IPFamily) {
+			libnet.SkipWhenClusterNotSupportIPFamily(ipFamily)
+
+			startPasstVMI := func(vmiBuilder func(opts ...libvmi.Option) *v1.VirtualMachineInstance, loginTo console.LoginToFunction) *v1.VirtualMachineInstance {
+				networkData, err := libnet.NewNetworkData(
+					libnet.WithEthernet("eth0",
+						libnet.WithDHCP4Enabled(),
+						libnet.WithDHCP6Enabled(),
+					),
+				)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				vmi := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithCloudInitNoCloudNetworkData(networkData),
+				)
+				vmi, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				vmi = libwait.WaitUntilVMIReady(vmi,
+					console.LoginToFedora,
+					libwait.WithFailOnWarnings(false),
+					libwait.WithTimeout(180),
+				)
+				return vmi
+			}
+
+			By("Starting a VMI")
+			migrateVMI := startPasstVMI(libvmi.NewFedora, console.LoginToFedora)
+			beforeMigNodeName := migrateVMI.Status.NodeName
+
+			By("Starting another VMI")
+			anotherVMI := startPasstVMI(libvmi.NewAlpine, console.LoginToAlpine)
+
+			By("Verify the VMIs can ping each other")
+			migrateVmiBeforeMigIP := libnet.GetVmiPrimaryIPByFamily(migrateVMI, ipFamily)
+			anotherVmiIP := libnet.GetVmiPrimaryIPByFamily(anotherVMI, ipFamily)
+			Expect(libnet.PingFromVMConsole(migrateVMI, anotherVmiIP)).To(Succeed())
+			Expect(libnet.PingFromVMConsole(anotherVMI, migrateVmiBeforeMigIP)).To(Succeed())
+
+			By("Perform migration")
+			migration := libmigration.New(migrateVMI.Name, migrateVMI.Namespace)
+			migrationUID := libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+			migrateVMI = libmigration.ConfirmVMIPostMigration(kubevirt.Client(), migrateVMI, migrationUID)
+
+			By("Verify all the containers in the source pod were terminated")
+			labelSelector := fmt.Sprintf("%s=%s", v1.CreatedByLabel, string(migrateVMI.GetUID()))
+			fieldSelector := fmt.Sprintf("spec.nodeName==%s", beforeMigNodeName)
+
+			Eventually(func() bool {
+				pods, err := kubevirt.Client().CoreV1().Pods(migrateVMI.Namespace).List(context.Background(),
+					metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(pods.Items)).To(Equal(1))
+
+				return pods.Items[0].Status.Phase == k8sv1.PodSucceeded
+			}, 30*time.Second).Should(BeTrue())
```

Comment:
- Please extract all these details out to a helper. e.g. `assertSourcePodContainsTerminate()`.
- Please avoid `Should(BeTrue())` assertions as they fail with no value information.
  Something like this will be better: [https://github.com/kubevirt/kubevirt/blob/b15ddb9eb942134a070d6bf7085d0956e9b0619e/tests/libmigration/migration.go#L398
  Adding a matcher for pod containers is too much and I am ](https://github.com/kubevirt/kubevirt/blob/f1e004fa31af23039ca5afba44f78e2fd00b8475/tests/vmi_configuration_test.go#L2351)
  (note that it replaces your eventually for the pod status check).

Reply:
* extract - done.
* BeTrue - done.